react-form-controls/my-app/src/RegistrationFormControlled.tsx

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+
+export function RegistrationFormControlled() {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    console.log('username', username);
+    console.log('password', password);
+  }
+  return (
+    <>
+      <form onSubmit={handleSubmit}>
+        <label>
+          Username:
+          <input
+            onChange={(e) => {
+              e.preventDefault();
```

Comment:
You don't need to prevent default in `onChange` handlers. The purpose of calling `preventDefault` is to keep the form from being submitted by the browser when the user clicks the submit button.